File: trunk/esper/src/test/java/net/esper/eql/parse/TestListParameter.java
package net.esper.eql.parse;

import junit.framework.TestCase;

import java.util.Set;

import net.esper.support.util.ArrayAssertionUtil;

public class TestListParameter extends TestCase
{
    private ListParameter listParam;

    public void setUp()
    {
        listParam = new ListParameter();
        listParam.add(new IntParameter(5));
        listParam.add(new FrequencyParameter(3));
    }

    public void testIsWildcard()
    {
        // Wildcard is expected to make only a best-guess effort, not be perfect
        assertTrue(listParam.isWildcard(5, 5));
        assertFalse(listParam.isWildcard(6, 10));
    }

    public void testGetValues()
    {
        Set<Integer> result = listParam.getValuesInRange(1, 8);
        ArrayAssertionUtil.assertEqualsAnyOrder(new int[] {3, 5, 6}, result);
    }
}

File: NEsper.Tests/eql/parse/TestListParameter.cs
using System;

using net.esper.compat;
using net.esper.support.util;

using NUnit.Core;
using NUnit.Framework;

namespace net.esper.eql.parse
{

    [TestFixture]
    public class TestListParameter
    {
        private ListParameter listParam;

        [SetUp]
        public virtual void setUp()
        {
            listParam = new ListParameter();
            listParam.Add(new IntParameter(5));
            listParam.Add(new FrequencyParameter(3));
        }

        [Test]
        public void testIsWildcard()
        {
            // Wildcard is expected to make only a best-guess effort, not be perfect
            Assert.IsTrue(listParam.IsWildcard(5, 5));
            Assert.IsFalse(listParam.IsWildcard(6, 10));
        }

        [Test]
        public void testGetValues()
        {
            Set<int> result = listParam.GetValuesInRange(1, 8);
            ArrayAssertionUtil.AreEqualAnyOrder(new int[] { 3, 5, 6 }, result);
        }
    }
}
